lib/srv/app/server.go

```diff
@@ -958,6 +958,7 @@ func (s *Server) authorizeContext(ctx context.Context) (*authz.Context, types.Ap
 		state,
 		matchers...)
 	if err != nil {
+		s.log.WithError(err).Warn("Converting to opaque access denied")
```

Comment:
Should the error message just say "Access denied to application %q"?

Reply:
Sure thing